hypnotoad/core/mesh.py

```diff
@@ -3213,6 +3269,21 @@ def f(psi, x):
             atol=atol,
             vectorized=True,
         )
+    except MaxIterException as e:
+        print("followPerpendicular failed: ", psirange, psivals, e.psi)
+        new_psivals = numpy.linspace(psi0, e.psi, len(psivals) + 1, endpoint=False)[1:]
+        return followPerpendicular(
+            None,
+            p0,
+            psi0,
+            f_R=f_R,
+            f_Z=f_Z,
+            psivals=new_psivals,
+            rtol=rtol,
+            atol=atol,
+            maxits=maxits,
+        )
```

Comment:
Would be useful to add a comment here to explain what happens (and why) when a `MaxIterException` is caught. At first glance it looks like this could build a grid that does not follow flux surfaces?

Reply:
Thanks @johnomotani Indeed this produces a grid that doesn't follow flux surfaces. Usually it's obvious visually when this has happened, but if run non-interactively then this could be a problem. Perhaps there should be a switch to determine whether MaxIterException terminates the mesh generation or not: By default it should terminate, but an interactive user might want to see a grid anyway.